devices/Rtc/Devices/Ds1302/Ds1302.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Device.Model;
+using Iot.Device.Common;
+
+namespace Iot.Device.Rtc
+{
+    /// <summary>
+    /// Realtime Clock Ds1302.
+    /// </summary>
+    [Interface("Realtime Clock Ds1302")]
+    public class Ds1302 : RtcBase, IDisposable
+    {
+        private GpioController _controller;
+        private GpioPin _clockPin;
+        private GpioPin _resetPin;
+        private GpioPin _dataPin;
+        private PinMode _currentDataPinDirection;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Ds1302"/> class.
+        /// </summary>
+        /// <param name="controller">Gpio controller.</param>
+        /// <param name="clockPin">Clock pin number.</param>
+        /// <param name="dataPin">Data pin number.</param>
+        /// <param name="resetPin">Reset pin number.</param>
+        /// <exception cref="ArgumentNullException">When controller was not provided.</exception>
+        public Ds1302(GpioController controller, int clockPin, int dataPin, int resetPin)
+        {
+            _controller = controller ?? throw new ArgumentNullException();
+            _clockPin = _controller?.OpenPin(clockPin, PinMode.Output);
```

Comment:
the controller can't be null here :-) you just checked it before ;-)

Reply:
on GpioPin, it's not part of the pattern to use them internally in bindings. But as we are introducing them to .NET IoT, I'm fine to keep them here. Otherwise, just use the pin numbers and it will work the same way